packages/dev/sharedUiComponents/src/lines/optionsLineComponent.tsx

```diff
@@ -80,44 +83,103 @@ export class OptionsLineComponent extends React.Component<IOptionsLineComponentP
     }
 
     updateValue(valueString: string) {
-        const value = this.props.valuesAreStrings ? valueString : parseInt(valueString);
+        let value = this.props.valuesAreStrings ? valueString : parseInt(valueString);
+
+        if (isNaN(Number(value))) {
+            for (let i = 0; i < this.props.options.length; i++) {
+                if (this.props.options.at(i)?.label === valueString) {
+                    value = Number(this.props.options.at(i)?.value);
+                }
+            }
+        }
+
+        this.forceUpdate();
+        if (value === 0) {
+            this.setState({ addCustom: true });
+        }
+
         this._localChange = true;
 
         const store = this.props.extractValue ? this.props.extractValue(this.props.target) : this.props.target[this.props.propertyName];
 
         if (!this.props.noDirectUpdate) {
             this.props.target[this.props.propertyName] = this._remapValueOut(value as number);
         }
-        this.setState({ value: value });
-
-        if (this.props.onSelect) {
-            this.props.onSelect(value);
+        if (value != 0) {

```

Comment:
Some things have caught my attention here. First, comparing to 0 using == (doesn't take types into account) instead of === (takes types into account) has some consequences:
```javascript
> false == 0
< true
> false === 0
< false
> '' == 0
< true
> '' === 0
< false
```
Is this intentional?

Second, I'm wondering about the need for this if. It means that we will never set the value 0 on the state?

Reply:
I did mean to take types into account. Would that be !== instead of != then?

And yes, we do not want the value of state to ever be 0. What the value of state is used for is to change the font to whatever font is correlated with that value. However, in my list, I have made the 'Custom Font' option correlated with the value 0. That means that without this check, when I click on Custom Font, the font of the selected text changes from, what it originally was to a default font (since Custom Font is not the name of a valid font). In order to prevent that from happening, if that value is 0, I simply do not change the value of state.

Reply:
> I did mean to take types into account. Would that be !== instead of != then?

Yes! 

Understand your reasoning now, thanks for explaining! 👍